openjdk-crac-17.yaml

```diff
@@ -0,0 +1,295 @@
+             "${{targets.subpkgdir}}"/usr/lib/jvm/java-17-openjdk
+
+  - name: "openjdk-crac-17-doc"
+    description: "OpenJDK 17 Documentation"
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/usr/lib/jvm/java-17-openjdk
+          mv "${{targets.destdir}}"/usr/lib/jvm/java-17-openjdk/man \
+             "${{targets.subpkgdir}}"/usr/lib/jvm/java-17-openjdk
+
+  - name: "openjdk-crac-17-demos"
+    description: "OpenJDK 17 Demos"
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/usr/lib/jvm/java-17-openjdk
+          mv "${{targets.destdir}}"/usr/lib/jvm/java-17-openjdk/demo \
+             "${{targets.subpkgdir}}"/usr/lib/jvm/java-17-openjdk
+
+  - name: "openjdk-crac-17-default-jvm"
+    description: "Use the openjdk-17 JVM as the default JVM"
+    dependencies:
+      runtime:
+        - openjdk-crac-17-jre
+      provides:
+        - default-crac-jvm=1.17
+        - default-crac-jvm-lts=1.17
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/usr/lib/jvm
+          ln -sf java-17-openjdk "${{targets.subpkgdir}}"/usr/lib/jvm/default-jvm
+
+  - name: "openjdk-crac-17-default-jdk"
+    description: "Use the openjdk-17 JVM as the default JVM with the JDK installed"
+    dependencies:
+      runtime:
+        - openjdk-crac-17-default-jvm
+        - openjdk-crac-17
+      provides:
+        - default-crac-jdk=1.17
+        - default-crac-jdk-lts=1.17
+
+update:
+  enabled: true
+  version-transform:
+    - match: ^(.+)\+(\d+)$
+      replace: $1.$2
+  github:
+    identifier: canonical/openjdk-17-crac
+    strip-prefix: jdk-
+    use-tag: true
+
+test:
+  environment:
+    contents:
+      packages:
+        - openjdk-crac-17-default-jdk
+        - openjdk-crac-17-jmods
+    environment:
+      JAVA_HOME: /usr/lib/jvm/java-17-openjdk
+  pipeline:
+    # Test a basic Hello World
+    - working-directory: basic
+      runs: |
+        javac HelloWorld.java
+        java HelloWorld | grep -qi "Hello World!"
+    # Test a basic HTTP connection
+    - working-directory: basic
+      runs: |
+        javac RequestTest.java
+        java RequestTest | grep -qi "Successfully connected to example.org"
+    # Test modules
+    - working-directory: advanced/module-project
+      runs: |
+        mkdir output
+        javac -d output --module-source-path modules $(find modules -name "*.java")
+
+        # Create a jar with the compiled classes
+        jar --verbose --create --file app.jar \
+          --main-class dev.chainguard.module.main.Main \
+          --module-version 1.0 \
+          -C output/test.modules . \
+          -C output/main.app .
+
+        # Test the jar
+        java -jar app.jar
+
+        # Test jlink
+        jlink --verbose --module-path "app.jar:$JAVA_HOME/jmods" \
+          --add-modules test.modules \
+          --output test-project-jre
+
+        # Test custom JRE
+        test-project-jre/bin/java -jar app.jar
```

Comment:
Are there any symlistic java crac-criu processes that we can checkpoint restart?

Something like "sleep 10; print hello world", start it, checkpoint it, kill it, restart it, observe that it comes back and prints hello world? As that's the point of this build that one can do this.

Reply:
Wrote a small Java program that does this, but I can't get a package test working here as invoking CRIU for checkpoint/restore requires it to be running in a container with elevated privileges (or the NET_ADMIN cap set). We'll need to cover this in our image tests. Testing with a privileged wolfi-base container with OpenJDK CRaC and CRaC CRIU layered currently

Reply:
fair enough!